src/test/pmemset_part/pmemset_part.c

```diff
@@ -2351,6 +2351,65 @@ test_pmemset_async_map_remove_multiple_part_maps(const struct test_case *tc,
 	return 3;
 }
 
+static struct pmemset *set_ptr = NULL;
+#define ARG_PTR ((void *)0xBADBADBAD)
+
+static int
+part_add_cb(struct pmemset *set, struct pmemset_event_context *ctx, void *arg)
+{
+	static int Counter;
+	UT_ASSERTeq(set, set_ptr);
+	UT_ASSERTeq(ctx->type, PMEMSET_EVENT_PART_ADD);
+	UT_ASSERTeq(arg, ARG_PTR);
+	return ++Counter;
+}
+
+/*
+ * test_part_map_set_event_part_add_cb - set a part add event callback in a
+ *                                       pmemset and map a part to this set
+ */
+static int
+test_part_map_set_event_part_add_cb(const struct test_case *tc, int argc,
+		char *argv[])
+{
+	if (argc < 1)
+		UT_FATAL("usage: test_part_map_set_event_part_add_cb <path>");
+
+	const char *file = argv[0];
+	struct pmemset_part *part;
+	struct pmemset_source *src;
+	struct pmemset *set;
+	struct pmemset_config *cfg;
+
+	int ret = pmemset_source_from_file(&src, file);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+
+	create_config(&cfg);
+
+	pmemset_config_set_event_callback(cfg, part_add_cb, ARG_PTR);
+
+	ret = pmemset_new(&set, cfg);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+
+	set_ptr = set;
+
+	ret = pmemset_part_new(&part, set, src, 0, 64 * 1024);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+
+	ret = pmemset_part_map(&part, NULL, NULL);
+	UT_ASSERTeq(ret, 1); /* callback function */
```

Comment:
What is this 1? (Please add comment) Is this return value from callback (Counter value)? If yes - isn't that strange we return callback ret val by api ret val? How can we know that the ret value is from func or from callback? If no - why this return non 0 value? I am a bit confused reading the documentation and looking at this test :)